test/test_operators.py

```diff
@@ -330,21 +330,34 @@ def linear_no_bias(input, weight):
 
     @TVMTest.given(
         shape=TVMTest.rand_shape(rank=2, min_dim=10),
-        out_features=TVMTest.rand_int(15, 30),
+        out_features=TVMTest.rand_int(15, 64),
     )
     def test_quantized_linear(self, shape, out_features):
+        # This is necessary since for N of size > 16 we enforce it to be
+        # multiple of 16. Right this defines packing of weights and that
+        # specifically makes this requirement necessary.
+        # On the other hand loosing this requirement needs changes that require
+        # some changes on relay side which otherwise complains during shape
```

Comment:
Can you elaborate the change? Or point out to some gist?

Reply:
In order to support arbitrary size on N we need to do:
1. Fix TVM. I will put up a "DONT COMMIT" PR for this. The reason we need to make this change is that right now if you try to do layout_transform in a way that dimension are not divisible, then things break because such transforms are not supported. It will silently do something else probably. For example.
Say you have weights: [10, 4]. layout_tranform from NK to NK16n4k means your new tensor is [1, 1, 16, 4]. Here dim[2] of 16 will have to be packed with zero for invalid elements. This does not happen today.
2. The REAL problem is this: Say we have a gemm of [1, 4]x[10,4]=[1,10]. Now after layout transform we will have our gemm produce [1, 16]. Post this we do requant. I rewrote requant to do take in [1, 16] and produce [1, 10] so that it only covers part of the tensor. In our quantized gemm implementation gemm and requant are part of the same op. Due to this relay complains during shape propagation that [1, 16] is larger than [1, 10]. A solution to this would be to split them up. We did not go that far and instead decided to rely on this assumption for now since for bert model most shapes are conformant.

Reply:
@llyfacebook, is that the answer you were looking for? Or your question was about something else?